infinispan-remote/src/main/java/org/hibernate/ogm/datastore/infinispanremote/impl/InfinispanRemoteDatastoreProvider.java

```diff
@@ -0,0 +1,192 @@
+/*
+ * Hibernate OGM, Domain model persistence for NoSQL datastores
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.ogm.datastore.infinispanremote.impl;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.TreeSet;
+
+import org.hibernate.ogm.datastore.infinispanremote.InfinispanRemoteDialect;
+import org.hibernate.ogm.datastore.infinispanremote.configuration.impl.InfinispanRemoteConfiguration;
+import org.hibernate.ogm.datastore.infinispanremote.impl.protobuf.SchemaDefinitions;
+import org.hibernate.ogm.datastore.infinispanremote.impl.protostream.ProtoDataMapper;
+import org.hibernate.ogm.datastore.infinispanremote.impl.protostream.ProtostreamSerializerSetup;
+import org.hibernate.ogm.datastore.infinispanremote.impl.schema.SequenceTableDefinition;
+import org.hibernate.ogm.datastore.infinispanremote.impl.sequences.HotRodSequenceHandler;
+import org.hibernate.ogm.datastore.infinispanremote.logging.impl.Log;
+import org.hibernate.ogm.datastore.infinispanremote.logging.impl.LoggerFactory;
+import org.hibernate.ogm.datastore.infinispanremote.spi.schema.SchemaCapture;
+import org.hibernate.ogm.datastore.infinispanremote.spi.schema.SchemaOverride;
+import org.hibernate.ogm.datastore.spi.BaseDatastoreProvider;
+import org.hibernate.ogm.datastore.spi.SchemaDefiner;
+import org.hibernate.ogm.dialect.spi.GridDialect;
+import org.hibernate.service.spi.Configurable;
+import org.hibernate.service.spi.ServiceRegistryAwareService;
+import org.hibernate.service.spi.ServiceRegistryImplementor;
+import org.hibernate.service.spi.Startable;
+import org.hibernate.service.spi.Stoppable;
+import org.infinispan.client.hotrod.RemoteCache;
+import org.infinispan.client.hotrod.RemoteCacheManager;
+import org.infinispan.protostream.SerializationContext;
+import org.infinispan.query.remote.client.ProtobufMetadataManagerConstants;
+
+/**
+ * @author Sanne Grinovero
+ */
+public class InfinispanRemoteDatastoreProvider extends BaseDatastoreProvider
+				implements Startable, Stoppable, Configurable, ServiceRegistryAwareService {
+
+	private static final Log log = LoggerFactory.getLogger();
+
+	// Only available during configuration
+	private InfinispanRemoteConfiguration config;
+
+	// The Hot Rod client; maintains TCP connections to the datagrid.
+	private RemoteCacheManager hotrodClient;
+
+	//Useful to allow people to dump the generated schema,
+	//we use it to capture the schema in tests too.
+	private SchemaCapture schemaCapture;
+
+	private ServiceRegistryImplementor serviceRegistry;
+
+	private SchemaOverride schemaOverrideService;
+
+	private Set<String> mappedCacheNames;
```

Comment:
`perCacheSchemaMappers.keySet()` is a subset of all mapped caches: we also use some for e.g. Sequences

Yet, I think you're right that I could change that: nothing would break as this is largely used just by testing utilities which don't care for Sequences.. but I like the explicitness of the Set in an explicit field, useful during debugging.